mesonbuild/dependencies.py

```diff
@@ -120,9 +120,12 @@ def __init__(self, name, environment, kwargs):
                     raise DependencyException('Pkg-config binary missing from cross file')
             else:
                 potential_pkgbin = environment.cross_info.config['binaries'].get('pkgconfig', 'non_existing_binary')
-                if shutil.which(potential_pkgbin):
+                rc = subprocess.call([potential_pkgbin, '--help'], stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL)
```

Comment:
Except that it would fail because the bug is that the `.exe` may be missing. So doing an `access('some_exe')` would fail because it would need to be `access('some_exe.exe')`. The only reliable and simple way to do this is to try to run it.